compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -37,28 +38,9 @@ def ensure(cls) -> CommandState:
         return click.get_current_context().ensure_object(CommandState)
 
 
-def init_endpoint_configuration_dir(conf_dir: pathlib.Path):
-    if not conf_dir.exists():
-        log.info("No existing configuration found at %s. Initializing...", conf_dir)
-        try:
-            conf_dir.mkdir(mode=0o700, exist_ok=True)
-        except Exception as exc:
```

Comment:
This brings up a discrepancy in how we've been creating the configuration directory.  As of somewhat recently, the endpoint sets the restrictive permissions, while the SDK hadn't.  They both use the same folder; suggest to add the mode to the SDK's variant.  This is nominally "the right thing to do," anyway.

Reply:
Err, no, _not_ the default.  Make it `0o700`, please.  Specifically not `0o777`.

Reply:
Works for me.